### PR TITLE
etcdctlv3/command: add with-prefix flag for get

### PR DESCRIPTION
**Pull request by siddontang**
Usage:
```
./bin/etcdctlv3 --endpoints=127.0.0.1:2378  get --with-prefix foo
```
It returns the data with key prefix "foo".

### CODE REVIEW

File: etcdctlv3/command/get_command.go [hunk 1]
```diff
@@ -40,8 +41,8 @@ func NewGetCommand() *cobra.Command {
 	cmd.Flags().StringVar(&getSortOrder, "order", "", "order of results; ASCEND or DESCEND")
 	cmd.Flags().StringVar(&getSortTarget, "sort-by", "", "sort target; CREATE, KEY, MODIFY, VALUE, or VERSION")
 	cmd.Flags().Int64Var(&getLimit, "limit", 0, "maximum number of results")
+	cmd.Flags().BoolVar(&getWithPrefix, "with-prefix", false, "enable keys with matching prefix")
```
Comment (xiang90) [inline etcdctlv3/command/get_command.go]: how about just `prefix`? (to be consistent with https://github.com/coreos/etcd/blob/master/etcdctlv3/command/watch_command.go#L43)

Comment (xiang90) [inline etcdctlv3/command/get_command.go]: also if prefix is set, we should check there is only one arg. (etcdctl foo foo10 --prefix does not make sense. foo10 is an additional arguement)

Comment (siddontang) [inline etcdctlv3/command/get_command.go]: got it.

**Comment by siddontang** [comment]
PTAL

File: etcdctlv3/command/get_command.go [hunk 2]
```diff
@@ -64,9 +65,11 @@ func getGetOp(cmd *cobra.Command, args []string) (string, []clientv3.OpOption) {
 
 	opts := []clientv3.OpOption{}
 	key := args[0]
-	if len(args) > 1 {
+	if len(args) > 1 && !getPrefix {
```
Comment (xiang90) [inline etcdctlv3/command/get_command.go]:
```go
if len(args) > 1  {
    if getPrefix {
        ExitWithError(ExitBadArgs, fmt.Errorf("too many arguments for range with prefix, only accept one."))
    }
    opts = append(opts, clientv3.WithRange(args[1]))
}
```
let's make this stricter.

**Comment by xiang90** [comment]
LGTM. Please squash all commits into one. Thanks for the effort!

**Comment by siddontang** [comment]
PTAL

**Pull request merged**